Content.Goobstation.Server/SlaughterDemon/Objectives/SlaughterDemonObjectiveSystem.cs

```diff
@@ -0,0 +1,180 @@
+            _meta.SetEntityName(ent.Owner, Loc.GetString(ent.Comp.Title), args.Meta);
+
+        if (ent.Comp.Description != null)
+            _meta.SetEntityDescription(ent.Owner, Loc.GetString(ent.Comp.Description), args.Meta);
+    }
+
+    private void OnAfterAssignObjective(Entity<SlaughterBaseObjectiveComponent> ent,
+        ref ObjectiveAfterAssignEvent args)
+    {
+        if (ent.Comp.Title != null)
+            _meta.SetEntityName(ent.Owner, Loc.GetString(ent.Comp.Title), args.Meta);
+
+        if (ent.Comp.Description != null)
+            _meta.SetEntityDescription(ent.Owner, Loc.GetString(ent.Comp.Description), args.Meta);
+    }
+
+    private void OnSpreadAssign(Entity<SlaughterSpreadBloodObjectiveComponent> ent, ref ObjectiveAssignedEvent args)
+    {
+        if (ent.Comp.Title == null)
+            return;
+
+        var areas = ent.Comp.Areas;
+        var randomArea = _random.Pick(areas);
+        var title = Loc.GetString(ent.Comp.Title, ("area", randomArea));
+
+        _meta.SetEntityName(ent.Owner, title);
+    }
+
+    private void OnAssignWizard(Entity<SlaughterKillTheWizardConditionComponent> ent, ref ObjectiveAssignedEvent args)
+    {
+        if (!TryComp<TargetObjectiveComponent>(ent.Owner, out var targetObjective))
+            return;
+
+        var query = EntityQueryEnumerator<VialSummonComponent>();
+        while (query.MoveNext(out _, out var comp))
+        {
+            if (comp.Used)
+                continue;
+
+            if (comp.Summoner == null)
+                continue;
+
+            _target.SetTarget(ent.Owner, comp.Summoner.Value, targetObjective);
+            comp.Used = true;
+            return;
+        }
+    }
+
+    private void OnGetBaseObjectiveProgress(Entity<SlaughterBaseObjectiveComponent> ent, ref ObjectiveGetProgressEvent args)
+    {
+        // fluff is fluff
+        args.Progress = 0.0f;
+    }
+
+    private void OnGetKillEveryoneProgress(Entity<SlaughterKillEveryoneConditionComponent> ent, ref ObjectiveGetProgressEvent args)
+    {
+        args.Progress = Progress(ent.Comp.Devoured, GetAllPlayers());
+    }
+
+    private void OnGetDevourProgress(Entity<SlaughterDevourConditionComponent> ent, ref ObjectiveGetProgressEvent args)
+    {
+        args.Progress = Progress(ent.Comp.Devour, _number.GetTarget(ent.Owner));
+    }
+
+    private void OnGetWizardKillProgress(Entity<SlaughterKillTheWizardConditionComponent> ent, ref ObjectiveGetProgressEvent args)
+    {
+        if (!_target.GetTarget(ent.Owner, out var targetUid))
+        {
+            args.Progress = 0f;
+            return;
+        }
+
+        args.Progress = GetWizardKillProgress(targetUid.Value);
+    }
+
+    private float GetWizardKillProgress(EntityUid target)
+    {
+        if (!_mind.TryGetMind(target, out _, out var mind))
+            return 1f;
+
+        var targetDead = _mind.IsCharacterDeadIc(mind);
+        if (!targetDead)
+            return 0f;
+
+        return 1f;
```

Comment:
```suggestion
        return !_mind.IsCharacterDeadIc(mind) ? 0f : 1f;
```
No point in extrapolating to an extra variable if you only use it once.